src/main/java/com/example/simpleannotation/model/GeneratedClass.java

```diff
@@ -0,0 +1,150 @@
+package com.example.simpleannotation.model;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Optional;
+
+import com.example.simpleannotation.exceptions.ClassGenerationException;
+import com.example.simpleannotation.model.descriptors.*;
+
+public abstract class GeneratedClass {
+    
+    private String packageName;
+    private String className;
+    private String extendClass;
+    private List<String> implementClasses;
+    private List<String> imports;
+    private List<String> staticImports;
+    private List<AttributeDescriptor> attributes;
+    private List<ConstructorDescriptor> constructors;
+    private List<MethodDescriptor> methods;
+
+    protected GeneratedClass () {
+        this.imports = new ArrayList<>();
+        this.staticImports = new ArrayList<>();
+        this.implementClasses = new ArrayList<>();
+        this.attributes = new ArrayList<>();
+        this.constructors = new ArrayList<>();
+        this.methods = new ArrayList<>();
+    }
+
+    public Optional<String> getPackageName() {
+        return Optional.ofNullable(packageName);
+    }
+
+    public GeneratedClass setPackageName(String packageName) {
+        this.packageName = packageName;
+        return this;
+    }
+
+    public Optional<String> getClassName() {
+        return Optional.ofNullable(className);
+    }
+
+    public GeneratedClass setClassName(String className) {
+        this.className = className;
+        return this;
+    }
+
+    public Optional<String> getExtendClass() {
+        return Optional.ofNullable(extendClass);
+    }
+
+    public GeneratedClass setExtendClass(String extendClass) {
+        this.extendClass = extendClass;
+        return this;
+    }
+
+    public List<String> getImplementClasses() {
+        return implementClasses;
+    }
+
+    public GeneratedClass setImplementClasses(List<String> implementClasses) {
+        this.implementClasses = implementClasses;
+        return this;
+    }
+
+    public List<String> getImports () {
```

Comment:
delete extra space  here and bellow

Reply:
Done. Revised in [pull request 8](https://github.com/OscarCamacho/simpleannotation/pull/8)